java/memory/memory-netty/src/test/java/org/apache/arrow/memory/TestBaseAllocator.java

```diff
@@ -1091,6 +1099,43 @@ public void testMemoryLeakWithReservation() throws Exception {
     }
   }
 
+  @Test
+  public void testMemoryUsage() {
+    ListAppender<ILoggingEvent> memoryLogsAppender = new ListAppender<>();
+    Logger logger = (Logger) LoggerFactory.getLogger("arrow.allocator");
+    try {
+      logger.setLevel(Level.TRACE);
+      logger.addAppender(memoryLogsAppender);
+      memoryLogsAppender.start();
+      try (ArrowBuf buf = new ArrowBuf(ReferenceManager.NO_OP, null,
+          1024, new PooledByteBufAllocatorL().empty.memoryAddress())) {
+        buf.memoryAddress();
+      }
+      BiFunction<List<String>, Level, Boolean> listAppenderContains =
+          (List<String> values, Level level) -> memoryLogsAppender.list.stream()
+              .anyMatch(
+                  log -> log.toString().contains(values.get(0)) &&
+                      log.toString().contains(values.get(1)) &&
+                      log.toString().contains(values.get(2)) &&
+                      log.getLevel().equals(level)
+              );
+      boolean result = false;
+      long startTime = System.currentTimeMillis();
+      while (true && (System.currentTimeMillis() - startTime) < 20000) {
+        result = listAppenderContains.apply(Arrays.asList("Memory Usage: \n", "Large buffers outstanding: ",
+                "Normal buffers outstanding: "), Level.TRACE);
+        if (result) {
+          break;
+        }
+      }
+      memoryLogsAppender.stop();
+      assertTrue(result);
+    } finally {
+      logger.detachAppender(memoryLogsAppender);
+      logger.setLevel(null);
```

Comment:
Shouldn't we reset it to what it originally was?

Reply:
Initially it was marked with null